### ISSUE TITLE
When -plain-http-tunneling is defined, no simulations are returned

### ISSUE THREAD

**Issue by oipat** [SpectoLabs/hoverfly#752]
Hi,

First and foremost thanks for the great tool!

I'm trying to use hoverfly to act as an HTTP MITM proxy. Our HTTP client uses HTTP tunneling and not directly proxied HTTP requests. I have a following simulation file:
```
{
  "data": {
    "pairs": [
      {
        "request": {
          "path": [
            {
              "matcher": "exact",
              "value": "/"
            }
          ],
          "method": [
            {
              "matcher": "exact",
              "value": "GET"
            }
          ],
          "destination": [
            {
              "matcher": "exact",
              "value": "example.org"
            }
          ],
          "scheme": [
            {
              "matcher": "exact",
              "value": "http"
            }
          ],
          "body": [
            {
              "matcher": "exact",
              "value": ""
            }
          ]
        },
        "response": {
          "status": 200,
          "body": "<!doctype html>\n<html>\n<head>\n    <title>Example Domain</title>\n\n    <meta charset=\"utf-8\" />\n    <meta http-equiv=\"Content-type\" content=\"text/html; charset=utf-8\" />\n    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\" />\n    <style type=\"text/css\">\n    body {\n        background-color: #f0f0f2;\n        margin: 0;\n        padding: 0;\n        font-family: \"Open Sans\", \"Helvetica Neue\", Helvetica, Arial, sans-serif;\n        \n    }\n    div {\n        width: 600px;\n        margin: 5em auto;\n        padding: 50px;\n        background-color: #fff;\n        border-radius: 1em;\n    }\n    a:link, a:visited {\n        color: #38488f;\n        text-decoration: none;\n    }\n    @media (max-width: 700px) {\n        body {\n            background-color: #fff;\n        }\n        div {\n            width: auto;\n            margin: 0 auto;\n            border-radius: 0;\n            padding: 1em;\n        }\n    }\n    </style>    \n</head>\n\n<body>\n<div>\n    <h1>Example Domain</h1>\n    <p>This domain is established to be used for illustrative examples in documents. You may use this\n    domain in examples without prior coordination or asking for permission.</p>\n    <p><a href=\"http://www.iana.org/domains/example\">More information...</a></p>\n</div>\n</body>\n</html>\n",
          "encodedBody": false,
          "headers": {
            "Accept-Ranges": [
              "bytes"
            ],
            "Cache-Control": [
              "max-age=604800"
            ],
            "Content-Length": [
              "1270"
            ],
            "Content-Type": [
              "text/html"
            ],
            "Date": [
              "Mon, 09 Jul 2018 17:16:40 GMT"
            ],
            "Etag": [
              "\"1541025663+gzip\""
            ],
            "Expires": [
              "Mon, 16 Jul 2018 17:16:40 GMT"
            ],
            "Hoverfly": [
              "Was-Here"
            ],
            "Last-Modified": [
              "Fri, 09 Aug 2013 23:54:35 GMT"
            ],
            "Server": [
              "ECS (lga/131A)"
            ],
            "Vary": [
              "Accept-Encoding"
            ],
            "X-Cache": [
              "HIT"
            ]
          },
          "templated": false
        }
      }
    ],
    "globalActions": {
      "delays": []
    }
  },
  "meta": {
    "schemaVersion": "v5",
    "hoverflyVersion": "v0.17.1",
    "timeExported": "2018-07-09T20:16:51+03:00"
  }
}
```

The simulation is not really relevant here though. I'm starting hoverfly with the following command: `hoverfly -spy -dest example.org -import /tmp/simulation.json -plain-http-tunneling -v`

When I curl with the following command:
`curl http://example.org --proxytunnel --proxy localhost:8500 -vvv`

Then hoverfly passes through the request to the real server instead of returning the simulation. And the weird thing is that if I start hoverfly without defining -dest, then it actually returns the simulation, but then it doesn't pass-through non-matching requests, for example google.com, and it just returns 502 with an error message.

So basically I'm trying use hoverfly simulations for certain host(s), and pass-through for other hosts. It doesn't seem to work as expected with -plain-http-tunneling and --proxytunnel requests. Without --proxytunnel it seems to work as expected.

Hoverfly version used: v0.17.1 (on OS X)

Let me know if you need more information regarding the issue.

**Comment by tommysitu** [comment]
Finally have a chance to look into this issue. Let me try to summarize the issue here:
1. `-plain-http-tunneling` with `-dest` will always pass through regardless the mode
2. `-plain-http-tunneling` without `-dest` try to simulate when it is supposed to spy. 

I've also found:
3. `-plain-http-tunneling` in capture mode without `-dest` failed with this error: 
```
ERRO[2018-11-28T23:54:12Z] There was an error when applying middleware to http request  error="parse ://example.org/: missing protocol scheme" mode=capture request="&map[method:GET scheme: destination:example.org path:/ query:map[] headers:map[User-Agent:[curl/7.54.0] Accept:[*/*]] body:]" response="&map[error:nil response]"
```

We are planning to fix this issue in the upcoming release.

**Comment by ttiurani** [comment]
Any progress on this issue @tommysitu ? It would be really helpful if there was a way to get passthrough to work also for --plain-http-tunneling.

**Comment by tommysitu** [comment]
@ttiurani It should be looked into next week.

**Comment by tommysitu** [comment]
@ttiurani did you experience the same issue? The problem only occurred if you make requests with `curl -proxytunnel`.

**Comment by ttiurani** [comment]
@tommysitu  This problem happens also with netty which needs the `--plain-http-tunneling` to get off the ground. IIUC the only way to mimic the way Netty does http connecting is to use this `--proxytunnel`.

So this is not curl specific at least. @oipat could shed some details on this?

**Comment by tommysitu** [comment]
Ok, I think I know what the issue is. Will push out more fix commits soonish.

**Comment by tommysitu** [comment]
PR https://github.com/SpectoLabs/hoverfly/pull/797 should fix all the above issues.
